backend/api/daemon/v1alpha/daemon.go

```diff
@@ -75,6 +77,58 @@ func (srv *Server) RegisterServer(rpc grpc.ServiceRegistrar) {
 	daemon.RegisterDaemonServer(rpc, srv)
 }
 
+// SyncResourcesWithPeer implements the corresponding gRPC method.
+func (srv *Server) SyncResourcesWithPeer(req *daemon.SyncResourcesWithPeerRequest, stream grpc.ServerStreamingServer[daemon.SyncingProgress]) error {
```

Comment:
@juligasa I'm not sure using streaming here is a good idea. If the client disconnects we probably don't want to cancel the syncing that was initiated, so it probably shouldn't be scoped to the API handler lifetime. WDYT?

Reply:
I think you are absolutely right in not attaching the the syncing lifecycle to the caller context. I just wanted to set a reasonable timeout and I did not know which one so I got the one from the call.
As for the usefulness of the stream I think it is useful. You can totally skip the streamed result and just look at the error return if you want. But frontend team has been complaining in the past that they wanted some kind of live feedback and this was quite cheap to add, so I imagine our new publish functionality like a progress bar on the bottom that eventually completes.